### PR TITLE
Restore full Windows tests

### LINKED ISSUE #17101: Restore full Windows tests
See this comment: https://github.com/pytorch/pytorch/pull/16649#discussion_r256702818

### PR DESCRIPTION
**Pull request by kostmo**
closes #17101

### CODE REVIEW

File: .jenkins/pytorch/win-test.sh [hunk 1]
```diff
@@ -74,89 +75,197 @@ set PYTHONPATH=%TMP_DIR_WIN%\\build;%PYTHONPATH%
...
+echo "KARL: A3"
+
 if ERRORLEVEL 1 exit /b 1
+
+echo "KARL: A4"
+
+
+:: Checking that torch is importable
+python -c "import torch"
+
```
Comment (peterjc123) [inline .jenkins/pytorch/win-test.sh]: nit: need one `if ERRORLEVEL 1 exit /b 1` here, otherwise the clause won't cause the whole script to fail.

**Comment by pjh5** [comment]
If this works then JOB_BASE_NAME must be saved in the docker images themselves, as AFAICT all usages of that variable in .circleci/ or .jenkins/ have been removed. We should find where that variable is coming from and make sure that it's encoded in BUILD_ENVIRONMENT correctly. 

I think this can be fixed by adding the "-test" + suffix to https://github.com/pytorch/ossci-job-dsl/blob/master/src/jobs/pytorch.groovy#L1070

**Comment by kostmo** [comment]
I think that `${JOB_BASE_NAME}` must currently still be nonempty.  Take a look at the console output of the [two "Test and Push" jobs](https://ci.pytorch.org/jenkins/job/pytorch-builds/job/pytorch-win-ws2016-cuda9-cudnn7-py3-trigger/40915/) from this PR's build.

In the console log of the `pytorch-win-ws2016-cuda9-cudnn7-py3-test1` job, one can find:

    22:50:00 + run_tests
    22:50:00 + '[' -z pytorch-win-ws2016-cuda9-cudnn7-py3-test1 ']'
    22:50:00 + [[ pytorch-win-ws2016-cuda9-cudnn7-py3-test1 == *-test ]]
    22:50:00 + [[ pytorch-win-ws2016-cuda9-cudnn7-py3-test1 == *-test1 ]]

Similarly, in the console log of the `pytorch-win-ws2016-cuda9-cudnn7-py3-test2` job, one can find:

    22:56:45 + [[ pytorch-win-ws2016-cuda9-cudnn7-py3-test2 == *-test2 ]]

**Comment by pjh5** [comment]
Can you tell where JOB_BASE_NAME is being populated from? It shouldn't exist anymore. All info should be encoded into BUILD_ENVIRONMENT.

**Comment by peterjc123** [comment]
@kostmo Would you please skip `DataLoaderTest.ChunkDataSetGetBatch` to see if it fixes the timeout issue?

**Comment by peterjc123** [comment]
Any updates on this PR? I believe we should fix this ASAP.

File: test/cpp/api/dataloader.cpp [hunk 2]
```diff
@@ -1603,6 +1603,7 @@ TEST(DataLoaderTest, StatefulDatasetWithCollate) {
   ASSERT_TRUE(batch->target[0].allclose(torch::zeros(kBatchSize - 1)));
 }
 
+/*
```
Comment (ezyang) [inline test/cpp/api/dataloader.cpp]: There should be a comment saying why this is commented out, and an issue tracking when we can uncomment it again.

But IIUC, you only wanted to disable this on Windows, right? Then it should be macro'ed.

**Review by ezyang — APPROVED** [review]
with comments

File: test/cpp/api/dataloader.cpp (hunk 2 above)
Comment (kostmo) [reply test/cpp/api/dataloader.cpp]: Are there pre-existing macros for this, or did you just mean to surround with `#ifdef`?

**Comment by kostmo** [comment]
Looks like disabling that test was not enough to fix it:
```18:29:39 test_many_tensors (__main__.TestTensorDataset) ... ok
18:29:39 test_single_tensor (__main__.TestTensorDataset) ... ok
18:29:39 
18:29:39 ======================================================================
18:29:39 FAIL: test_proper_exit (__main__.TestDataLoader)
18:29:39 There might be ConnectionResetError or leaked semaphore warning (due to dirty process exit), but they are all safe to ignore
18:29:39 ----------------------------------------------------------------------
18:29:39 Traceback (most recent call last):
18:29:39   File "C:\Jenkins\workspace\pytorch-builds\pytorch-win-ws2016-cuda9-cudnn7-py3-test2\test\common_utils.py", line 120, in wrapper
18:29:39     fn(*args, **kwargs)
18:29:39   File "test_dataloader.py", line 757, in test_proper_exit
18:29:39     self.fail(fail_msg + ', and had exception {}'.format(loader_p.exception))
18:29:39 AssertionError: test_proper_exit with use_workers=True, pin_memory=True, hold_iter_reference=True, exit_method=None: loader process failed to setup within given time, and had exception Traceback (most recent call last):
18:29:39   File "C:\Jenkins\workspace\pytorch-builds\pytorch-win-ws2016-cuda9-cudnn7-py3-test2\test\test_dataloader.py", line 175, in run
18:29:39     super(ErrorTrackingProcess, self).run()
18:29:39   File "C:\Jenkins\Miniconda3\lib\multiprocessing\process.py", line 93, in run
18:29:39     self._target(*self._args, **self._kwargs)
18:29:39   File "C:\Jenkins\workspace\pytorch-builds\pytorch-win-ws2016-cuda9-cudnn7-py3-test2\test\test_dataloader.py", line 354, in _test_proper_exit
18:29:39     for i, _ in enumerate(it):
18:29:39   File "C:\Jenkins\workspace\pytorch-builds\pytorch-win-ws2016-cuda9-cudnn7-py3-test2\build\win_tmp\build\torch\utils\data\dataloader.py", line 545, in __next__
18:29:39     idx, batch = self._get_batch()
18:29:39   File "C:\Jenkins\workspace\pytorch-builds\pytorch-win-ws2016-cuda9-cudnn7-py3-test2\build\win_tmp\build\torch\utils\data\dataloader.py", line 517, in _get_batch
18:29:39     raise RuntimeError('Pin memory thread exited unexpectedly')
18:29:39 RuntimeError: Pin memory thread exited unexpectedly
18:29:39 
18:29:39 
18:29:39 ----------------------------------------------------------------------
18:29:39 Ran 51 tests in 73.087s
18:29:39 
18:29:39 FAILED (failures=1, skipped=2)
18:29:39 Traceback (most recent call last):
18:29:39   File "run_test.py", line 458, in <module>
18:29:39     main()
18:29:39   File "run_test.py", line 450, in main
18:29:39     raise RuntimeError(message)
18:29:39 RuntimeError: test_dataloader failed!
```

**Comment by peterjc123** [comment]
@kostmo Looks like `test_proper_exit` is only a flaky test. But it gets stuck at `DataLoaderTest.ChunkDataSetWithEmptyBatch` now. I think that the implementation of `ChunkDataSet` 
 doesn't work well and causes deadlock on Windows.

**Comment by peterjc123** [comment]
Opened a new issue for the deadlock: #17609.

**Pull request closed without merging**